nesta/packages/arxiv/tests/test_collect_arxiv.py

```diff
@@ -32,188 +31,190 @@ def mock_response():
+        batch, _ = arxiv_batch('111222444|1')
+        expected_fields = {'datestamp', 'id', 'created', 'updated', 'title', 'categories',
+                           'journal_ref', 'doi', 'msc_class', 'abstract', 'authors'}
+        assert set(batch[0]) == expected_fields
+
+    @mock.patch('nesta.packages.arxiv.collect_arxiv._arxiv_request', autospec=True)
+    def test_arxiv_batch_handles_missing_fields(self, mocked_request, mock_response):
+        mocked_request.return_value = ET.fromstring(mock_response)
+        batch, _ = arxiv_batch('111222444|1')
+        expected_fields = {'datestamp', 'id', 'title', 'categories', 'abstract', 'authors'}
+        assert set(batch[1]) == expected_fields
+
+    @mock.patch('nesta.packages.arxiv.collect_arxiv._arxiv_request', autospec=True)
+    def test_arxiv_batch_author_json_conversion(self, mocked_request, mock_response):
+        mocked_request.return_value = ET.fromstring(mock_response)
+        batch, _ = arxiv_batch('111222444|1')
+
+        expected_authors = '[{"keyname": "Author", "forenames": "G."}]'
+        assert batch[0]['authors'] == expected_authors
+
+        expected_authors = ('[{"keyname": "AnotherAuthor", "forenames": "L. M.", "suffix": "II", '
+                            '"affiliation": "CREST, Japan Science and Technology Agency"}, '
+                            '{"keyname": "Surname", "forenames": "Some other"}, '
+                            '{"keyname": "Collaboration", "forenames": "An important"}]')
+        assert batch[1]['authors'] == expected_authors
+
+    @mock.patch('nesta.packages.arxiv.collect_arxiv._arxiv_request', autospec=True)
+    def test_arxiv_batch_converts_categories_to_list(self, mocked_request, mock_response):
+        mocked_request.return_value = ET.fromstring(mock_response)
+        batch, _ = arxiv_batch('111222444|1')
+        assert batch[0]['categories'] == ['math.PR', 'math.RT']
+        assert batch[1]['categories'] == ['hep-ex']
+
+    @mock.patch('nesta.packages.arxiv.collect_arxiv._arxiv_request', autospec=True)
+    def test_arxiv_batch_converts_dates(self, mocked_request, mock_response):
+        mocked_request.return_value = ET.fromstring(mock_response)
+        batch, _ = arxiv_batch('111222444|1')
+        assert {'datestamp', 'created', 'updated'} < batch[0].keys()
+        assert {'created', 'updated'}.isdisjoint(batch[1].keys())
+
+    @mock.patch('nesta.packages.arxiv.collect_arxiv._arxiv_request', autospec=True)
+    def test_arxiv_batch_returns_resumption_cursor(self, mocked_request, mock_response):
+        mocked_request.return_value = ET.fromstring(mock_response)
+        _, reumption_token = arxiv_batch('111222444|1')
+        assert reumption_token == '3132962|1001'
+
+    @mock.patch('nesta.packages.arxiv.collect_arxiv._arxiv_request', autospec=True)
+    def test_arxiv_batch_returns_none_at_end_of_data(self, mocked_request):
+        mock_response = b'''<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
I would pop this dummy response in a file and retrieve it with a fixture

Reply:
as it's only used once I've kept this in as I think it's helpful to see the input with the test, but I was able to halve the amount of xml (there was a lot of redundancy) so now it's comparable to the other blocks in this test file.